Deux erreurs de PEP8 (standard d'écriture) à corriger:

```
plateforme_budgetaire/project/views.py:131:80: E501 line too long (82 > 79 characters)
    subproject = get_object_or_404(SubProject, id=self.kwargs['subProjectID'])
```

```
plateforme_budgetaire/project/views.py:138:80: E501 line too long (90 > 79 characters)
    'Vous ne disposer des droits nécessaire afin de supprimer ce sous-projet!'
```

Pour cela tu peux simplement faire :
- Correction de tes erreurs 
- Git add ton_fichier
- Git commit --amend
- Git push -f origin ta_branche

Cela va ajouter les derniers changements tracker (git add) au dernier commit.

--- a/plateforme_budgetaire/project/templates/project/project_detail.html
+++ b/plateforme_budgetaire/project/templates/project/project_detail.html
@@ -54,6 +54,7 @@
             {% for subProject in project.SubProject.all %}
             <div class="panel panel-default">
                 <div class="panel-heading">
+                    <a class="btn-xs btn-danger pull-right" href='{% url "projects:subproject_delete" subProject.id %}' role="button" >X</a>
                     <h4 class="panel-title">
                         {{ subProject.name }}
                     </h4>
@@ -76,12 +77,12 @@
     ------------>
 
     <!-- Accept modal -->
-    <div class="modal fade" id="AcceptModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+    <div class="modal fade" id="AcceptModal" tabindex="-1" role="dialog" aria-labelledby="acceptlLabel">
       <div class="modal-dialog" role="document">
         <div class="modal-content">
           <div class="modal-header">
             <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
-            <h4 class="modal-title" id="myModalLabel">Accepter le projet</h4>
+            <h4 class="modal-title" id="acceptlLabel">Accepter le projet</h4>
           </div>
           <div class="modal-body">
             Êtes-vous sûr de vouloir <span class="text-success"><strong>accepter</strong></span> le projet <strong>{{project.name}}</strong>?
@@ -95,12 +96,12 @@
     </div>
 
     <!-- Refuse Modal -->
-    <div class="modal fade" id="RefuseModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+    <div class="modal fade" id="RefuseModal" tabindex="-1" role="dialog" aria-labelledby="refuseLabel">
       <div class="modal-dialog" role="document">
         <div class="modal-content">
           <div class="modal-header">
             <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
-            <h4 class="modal-title" id="myModalLabel">Rejeter le projet</h4>
+            <h4 class="modal-title" id="refuseLabel">Rejeter le projet</h4>
           </div>
           <div class="modal-body">
             Êtes-vous sûr de vouloir <span class="text-danger"><strong>rejeter</strong></span> le projet <strong>{{project.name}}</strong>?

--- a/plateforme_budgetaire/project/templates/project/subproject_delete.html
+++ b/plateforme_budgetaire/project/templates/project/subproject_delete.html
@@ -1,0 +1,17 @@
+{% extends "landing_base.html" %}
+
+{% block extrajs %}
+
+{% endblock %}
+
+{% block content %}
+    <div class="row">
+        <strong>Êtes-vous sûr de vouloir supprimer le sous-projet "{{subProject.name}}" du projet "{{subProject.project}}" ? </strong>
+            <form method="post" action="">
+                {% csrf_token %}
+                <button class="btn btn-danger" type="submit" >Oui</button>
+                <a class="btn btn-warning" href='{% url "projects:project_detail" subProject.project.id %}'>Non</a>
+            </form>
+        </div>
+    </div>
+{% endblock %}

--- a/plateforme_budgetaire/project/urls.py
+++ b/plateforme_budgetaire/project/urls.py
@@ -29,6 +29,11 @@ urlpatterns = patterns(
         SubProjectCreate.as_view(),
         name='subproject_create'
     ),
+    url(
+        r'^subDelete/(?P<subProjectID>\d+)$',
+        SubProjectDelete.as_view(),
+        name='subproject_delete'
+    ),
 
     # Function
     url(

--- a/plateforme_budgetaire/project/views.py
+++ b/plateforme_budgetaire/project/views.py
@@ -5,6 +5,7 @@ from project.models import Project, SubProject, PROJECT_STATUS_CHOICES
 from project.forms import ProjectsForm, SubProjectsForm
 from django.core.urlresolvers import reverse_lazy
 from django.shortcuts import redirect
+from django.shortcuts import get_object_or_404
 from django.utils.decorators import method_decorator
 from django.contrib.auth.decorators import login_required
 from django.contrib import messages
@@ -114,6 +115,44 @@ class SubProjectCreate(generic.CreateView):
             self.request,
             messages.SUCCESS,
             'Le sous-projet à bien été ajouté!'
+        )
+        return self.object.get_absolute_url()
+
+
+class SubProjectDelete(generic.DeleteView):
+    form_class = SubProjectsForm
+    context_object_name = 'subProject'
+    template_name = 'project/subproject_delete.html'
+    model = SubProject
+    pk_url_kwarg = 'subProjectID'
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        subproject = get_object_or_404(
+            SubProject,
+            id=self.kwargs['subProjectID']
+        )
+        is_creator = self.request.user == subproject.project.creator
+        is_staff = self.request.user.is_staff
+
+        if is_staff or is_creator:
+            return super(SubProjectDelete, self).dispatch(*args, **kwargs)
+        else:
+            messages.add_message(
+                self.request,
+                messages.ERROR,
+                'Vous ne disposer des droits nécessaire'
+                ' afin de supprimer ce sous-projet!'
+            )
+            return redirect(reverse_lazy(
+                "projects:project_list"
+            ))
+
+    def get_success_url(self):
+        messages.add_message(
+            self.request,
+            messages.SUCCESS,
+            'Le sous-projet à bien été supprimé!'
         )
         return self.object.get_absolute_url()
 

